libr/bin/p/bin_le.c

```diff
@@ -0,0 +1,166 @@
+/* radare - LGPL - Copyright 2019 - GustavoLCR */
+
+#include <r_bin.h>
+#include "../format/le/le.h"
+
+static bool check_buffer(RBuffer *b) {
+	ut64 length = r_buf_size (b);
+	if (length <= 0x3d) {
+		return false;
+	}
+	ut16 idx = r_buf_read_le16_at (b, 0x3c);
+	if ((ut64)idx + 26 < length) {
+		ut8 buf[2];
+		r_buf_read_at (b, 0, buf, sizeof (buf));
+		if (!memcmp (buf, "LX", 2) || !memcmp (buf, "LE", 2)) {
+			return true;
+		} else if (!memcmp (buf, "MZ", 2)) {
+			r_buf_read_at (b, idx, buf, sizeof (buf));
+			if (!memcmp (buf, "LX", 2) || !memcmp (buf, "LE", 2)) {
+				return true;
+			}
+		}
+	}
+	return false;
+}
+
+static bool load_buffer(RBinFile *bf, void **bin_obj, RBuffer *buf, ut64 loadaddr, Sdb *sdb) {
+	r_return_val_if_fail (bf && bin_obj && buf, false);
+	r_bin_le_obj_t *res = r_bin_le_new_buf (buf);
+	if (res) {
+		*bin_obj = res;
+		return true;
+	}
+	return false;
+}
+
+static void destroy(RBinFile *bf) {
+	r_bin_le_free (bf->o->bin_obj);
+}
+
+static void header(RBinFile *bf) {
+	RBin *rbin = bf->rbin;
+	r_bin_le_obj_t *bin = bf->o->bin_obj;
```

Comment:
maybe do some nullchks and do PrintfCallback p = rbin->cb_printf to make further calls non indirect